File: BE/MIS_Project_Java/src/main/java/com/example/demo/LoaiSanPham.java
package com.example.demo;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.ArrayList;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;


@JsonIgnoreProperties(value = { "all" })
public class LoaiSanPham {
	public String maLoaiSP;
	public String tenLoaiSP;
	public String maNhomSP;
	public String tenNhomSP;
	
	public LoaiSanPham()
	{
		this.maLoaiSP = "";
		this.tenLoaiSP = "";
		this.maNhomSP = "";
		this.tenNhomSP = "";
	}
	
	public LoaiSanPham(String MaLoaiSP, String TenLoaiSP, String MaNhomSP, String TenNhomSP)
	{
		this.maLoaiSP = MaLoaiSP;
		this.tenLoaiSP = TenLoaiSP;
		this.maNhomSP = MaNhomSP;
		this.tenNhomSP = TenNhomSP;
	}
	
	public List<LoaiSanPham> getAll()
	{
		List<LoaiSanPham> typeProducts = new ArrayList<LoaiSanPham>();
		try 
		 {
			 String dbURL = "jdbc:sqlserver://localhost;databaseName=MIS;user=sa;password=t";
		     Connection conn = DriverManager.getConnection(dbURL);
		     if (conn != null) 
		     {
		         Statement stmt = conn.createStatement();
	             ResultSet rs = stmt.executeQuery("SELECT * FROM SanPham SP, LoaiSanPham LSP, NhomSanPham NSP, CuaHang CH WHERE SP.loaiSP = LSP.maLoaiSP and LSP.maNhomSP = NSP.maNhomSP and CH.AccountID = SP.Account_CH");
	             while (rs.next()) 
	             {
	            	 LoaiSanPham typeProduct = new LoaiSanPham(rs.getString(1), rs.getString(2), rs.getString(3), rs.getString(5));
	            	 typeProducts.add(typeProduct);
	             }
	             conn.close();
	             return typeProducts;
		     }
		 } 
		 catch (SQLException ex) 
		 {
			 System.err.println("Cannot connect database, " + ex);
		 }
		 	 return null;
	}
	
	public List<LoaiSanPham> getMany(String cateID)
	{
		List<LoaiSanPham> typeProducts = new ArrayList<LoaiSanPham>();
		try 
		 {
			 String dbURL = "jdbc:sqlserver://localhost;databaseName=MIS;user=sa;password=t";
		     Connection conn = DriverManager.getConnection(dbURL);
		     if (conn != null) 
		     {
		         Statement stmt = conn.createStatement();
	             ResultSet rs = stmt.executeQuery("SELECT * FROM LoaiSanPham LSP, NhomSanPham NSP WHERE LSP.maNhomSP = NSP.maNhomSP and NSP.maNhomSP = '" + cateID + "'");
	             while (rs.next()) 
	             {
	            	 LoaiSanPham typeProduct = new LoaiSanPham(rs.getString(1), rs.getString(2), rs.getString(3), rs.getString(5));
	            	 typeProducts.add(typeProduct);
	             }
	             conn.close();
	             return typeProducts;
		     }
		 } 
		 catch (SQLException ex) 
		 {
			 System.err.println("Cannot connect database, " + ex);
		 }
		 	 return null;
	}

}

File: BE/MIS_Project_C/MIS_Project/Models/LoaiSanPham.cs
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Threading.Tasks;

namespace MIS_Project.Models
{
    public class LoaiSanPham
    {
        public string MaLoaiSP { get; set; }
        public string TenLoaiSP { get; set; }
        public string MaNhomSP { get; set; }
        public string TenNhomSP { get; set; }


        public List<LoaiSanPham> getAll()
        {
            SqlCommand com = new SqlCommand();
            SqlDataReader dr;
            SqlConnection con = new SqlConnection();
            con.ConnectionString = MIS_Project.Properties.Resources.ConnectionString;
            List<LoaiSanPham> typeProducts = new List<LoaiSanPham>();
            try
            {
                con.Open();
                com.Connection = con;
                com.CommandText = "SELECT * FROM LoaiSanPham LSP, NhomSanPham NSP WHERE LSP.MaNhomSP = NSP.MaNhomSP";
                dr = com.ExecuteReader();
                while (dr.Read())
                {
                    typeProducts.Add(new LoaiSanPham()
                    {
                        MaLoaiSP = dr["MaLoaiSP"].ToString(),
                        TenLoaiSP = dr["TenLoaiSP"].ToString(),
                        MaNhomSP = dr["MaNhomSP"].ToString(),
                        TenNhomSP = dr["TenNhomSP"].ToString()
                    });
                }
                con.Close();
            }
            catch (Exception exc)
            {
                throw exc;
            }
            return typeProducts;
        }

        public List<LoaiSanPham> getMany(string CateID)
        {
            SqlCommand com = new SqlCommand();
            SqlDataReader dr;
            SqlConnection con = new SqlConnection();
            con.ConnectionString = MIS_Project.Properties.Resources.ConnectionString;
            List<LoaiSanPham> typeProducts = new List<LoaiSanPham>();
            try
            {
                con.Open();
                com.Connection = con;
                com.CommandText = "SELECT * FROM LoaiSanPham LSP, NhomSanPham NSP WHERE LSP.maNhomSP = NSP.maNhomSP and NSP.maNhomSP = '" + CateID + "'";
                dr = com.ExecuteReader();
                while (dr.Read())
                {
                    typeProducts.Add(new LoaiSanPham()
                    {
                        MaLoaiSP = dr["MaLoaiSP"].ToString(),
                        TenLoaiSP = dr["TenLoaiSP"].ToString(),
                        MaNhomSP = dr["MaNhomSP"].ToString(),
                        TenNhomSP = dr["TenNhomSP"].ToString()
                    });
                }
                con.Close();
            }
            catch (Exception exc)
            {
                throw exc;
            }
            return typeProducts;
        }
    }
}
